backend/app/api/endpoints/users.py

```diff
@@ -15,10 +19,33 @@ async def message_users():
     return {"message": "Hello User!!!"}
 
 
+@router.post("/verify-openid")
+async def verify_openid(request: Request, open_id_info: OpenIdInfo):
+    matrix_homeserver = "https://" + open_id_info.matrix_server_name
+    params = {"access_token": open_id_info.access_token}
+    url = urljoin(matrix_homeserver, "/_matrix/federation/v1/openid/userinfo")
+
+    try:
+        async with http_client.session.get(url, params=params) as resp:
+            if resp.status != 200:
+                raise HTTPException(status_code=404, detail="Invalid token")
+            data = await resp.json()
+
+            server_session_data = ServerSessionData(matrix_user=data["sub"])
+
+            response = JSONResponse({"message": "success"})
+            # Creating a server session with the matrix username.
+            response = fastapi_sessions.create_session(response, data=server_session_data)
+            return response
+
+    except aiohttp.ClientConnectionError as err:
+        raise HTTPException(status_code=500, detail="Internal server error")
```

Comment:
We should add logging, specifically for stuff like this, we are going to need to know why it died here.